### DISCUSSION TITLE
IDEA: Unreal engine styled comments

### DISCUSSION THREAD

**Discussion by Datboi449** [ScratchAddons/ScratchAddons#5757] [Ideas]
In unreal engine, which has a similar coding interface to scratch, you can create giant transparent blocks to be displayed in the background of the code editor. Though it's not great for notes/info, it is spectacular for organizing code into different sections if needed. This would be incredible useful if you have a sprite that contains many clones that do different things.  

CONCEPT:
![scratchIdea](https://user-images.githubusercontent.com/93498441/221450700-6972fc9d-f72a-459e-86a4-f63e0623816b.png)
UNREAL ENGINE:
![NodeActions](https://user-images.githubusercontent.com/93498441/221450711-31ae33ec-1b0b-426a-b72c-5a0fd8cb0bb8.jpg)

**Comment by DNin01** [comment]
We'd need to find a way to make this not break for anyone who isn't using Scratch Addons. For example, `sprite-folders` changes the name of sprites in your project behind the scenes, which gets stored in the project's files. It'll look a little weird to people who aren't using SA, but it doesn't ruin anything. We could do something similar for the comments, though, if it looks too cryptic or strange when not using SA, it might not be worth it.

**Reply by CST1229** [reply]
It could be a comment that begins with "Group: " and is behind other scripts.

**Reply by nexensys** [reply]
It could also change the ID of blocks... that's an idea

**Reply by CST1229** [reply]
> It could also change the ID of blocks... that's an idea

That would probably increase filesizes and also not work well with project JSON compression.
